src/utils/filter-utils.js

```diff
@@ -563,3 +664,183 @@ export function getDefaultFilterPlotType(filter) {
 
   return filterPlotTypes[filter.yAxis.type] || null;
 }
+
+/**
+ *
+ * @param datasetIds list of dataset ids
+ * @param datasets
+ * @param filters
+ * @return {{[p: string]: *}}
+ */
+export function applyFiltersToDatasets(datasetIds, datasets, filters) {
+  const dataIds = Array.isArray(datasetIds) ? datasetIds : [datasetIds];
```

Comment:
Could collect a unique list of `datasetId` based on filters passed. So that we don't need to filter through all datasets

Reply:
nice